deploy/examples/create-external-cluster-resources.py

```diff
@@ -34,7 +32,8 @@
 if sys.version_info.major >= 3:
```

Comment:
The error actually comes from here, you see if we would call this file with python2 then the "variables" in this case they are a file function and a class, so maybe that error message could indeed be improved.
But I think the style error still is valid so with the example below for example will throw error

```
if True == False:
    foo = "bar"
print(foo)
```
The easiest way to fix this is to give the foo a baseline value and the style error and the potential bug is removed

```
foo = ""
if True == False:
    foo = "bar"
print(foo)
```

However, in this particular example it is a bit harder as the [ipaddress library](https://docs.python.org/3/library/ipaddress.html) was added in python 3.3

So that was why my suggestion in the issue was to completely drop python2 and that would fix the style issue and make the code overall more readable when simply removing the if-statement check for python version. Of course the "variables" could also be defined as None, but then the python2 code is not working anyways. I suppose this code cannot at this moment either work with python2 as these two variables are not defined but maybe it is partially working :smile: 

The other option I would consider if for some reason we care about supporting python2 is rewriting this functions for example using the sockets library

Hope this is helpful!

Reply:
We want to keep the python2 support

Reply:
Okay, interesting. I think Python 2 probably should be part of the tests somehow then also otherwise this is very easy to break.

I think the best plan of action would be to remove the dependencies on ipaddress and use socket instead. Something like socket.inet_pton(socket.AF_INET, ip_str) should be able to determine the IPv type. What do you think about this approach, @parth-gr? Do you think this would be a feasible fix?

I also found this as an option: https://github.com/phihag/ipaddress, but I think that would be quite hard to make explicit as there is no real dependency management, especially in the Python 2 case.

Others might of course have some other opinions on how this should be fixed, which are more than welcome. 😄

Reply:
@parth-gr IMHO dropping Python2 support is something you may seriously consider as it's has been EOL for more than 4 years: https://devguide.python.org/versions/. All modern distros come with Python3 so I don't see the need to stick to a very old version with  is already EOL.

Reply:
Could you remind me why we want to keep py2 support? Supporting only py3 would vastly simplify things.

IIRC, this had something to do with some Ceph versions/containers that still relied on py2. However, it's been a while since we talked about this, and things have a tendency to change over time. It's possible this isn't as necessary anymore, and this seems like the right time to revisit the discussion.

Reply:
Yes it is mainly depend on the RHCS version, what it supports